packages/http-client-csharp/emitter/src/lib/example-converter.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import {
+  SdkAnyExample,
+  SdkArrayExample,
+  SdkBooleanExample,
+  SdkContext,
+  SdkDictionaryExample,
+  SdkHttpOperationExample,
+  SdkHttpParameter,
+  SdkHttpParameterExample,
+  SdkHttpResponse,
+  SdkHttpResponseExample,
+  SdkModelExample,
+  SdkNullExample,
+  SdkNumberExample,
+  SdkStringExample,
+  SdkTypeExample,
+  SdkUnionExample,
+} from "@azure-tools/typespec-client-generator-core";
+import { NetEmitterOptions } from "../options.js";
+import {
+  InputAnyExample,
+  InputArrayExample,
+  InputBooleanExample,
+  InputDictionaryExample,
+  InputHttpOperationExample,
+  InputModelExample,
+  InputNullExample,
+  InputNumberExample,
+  InputParameterExample,
+  InputStringExample,
+  InputTypeExample,
+  InputUnionExample,
+  OperationResponseExample,
+} from "../type/input-examples.js";
+import { InputParameter } from "../type/input-parameter.js";
+import {
+  InputArrayType,
+  InputDictionaryType,
+  InputModelType,
+  InputNullableType,
+  InputPrimitiveType,
+  InputUnionType,
+} from "../type/input-type.js";
+import { OperationResponse } from "../type/operation-response.js";
+import { SdkTypeMap } from "../type/sdk-type-map.js";
+import { fromSdkType } from "./converter.js";
+
+export function fromSdkHttpExamples(
+  sdkContext: SdkContext<NetEmitterOptions>,
+  examples: SdkHttpOperationExample[] | undefined,
+  parameterMap: Map<SdkHttpParameter, InputParameter>,
+  responseMap: Map<SdkHttpResponse, OperationResponse>,
+  typeMap: SdkTypeMap
+): InputHttpOperationExample[] | undefined {
+  if (!examples) return undefined;
+
+  return examples.map((example) => fromSdkHttpExample(example));
+
+  function fromSdkHttpExample(example: SdkHttpOperationExample): InputHttpOperationExample {
```

Comment:
Emm I put these function in closure because they are using a lot of the context variables from above, such as the `typeMap`, `context`. I do not want these methods to individually defines those parameters therefore I put them in this closure so that we could omit them.
It might be more clear because this function is the only place that these functions should be used